src/hotspot/share/gc/parallel/parallelScavengeHeap.cpp

```diff
@@ -676,6 +672,32 @@ void ParallelScavengeHeap::verify(VerifyOption option /* ignored */) {
   }
 }
 
+void ParallelScavengeHeap::trace_actual_reserved_page_size(const size_t reserved_heap_size, const ReservedSpace rs) {
+  // Check if Info level is enabled, since os::trace_page_sizes() logs on Info level.
+  if(log_is_enabled(Info, pagesize)) {
+    size_t page_size = os::vm_page_size();
+    if (UseLargePages) {
+      // There are two ways to manage large page memory.
+      // 1. OS supports committing large page memory.
+      // 2. OS doesn't support committing large page memory so ReservedSpace manages it.
+      //    And ReservedSpace calls it 'special'. If we failed to set 'special',
+      //    we reserved memory without large page.
+      if (os::can_commit_large_page_memory() || rs.special()) {
+        // An alignment at ReservedSpace comes from preferred page size or
+        // heap alignment, and if the alignment came from heap alignment, it could be
+        // larger than large pages size. So need to cap with the large page size.
+        page_size = MIN2(rs.alignment(), os::large_page_size());
+      }
+    }
```

Comment:
Since this is exactly the same code as in `actual_reserved_page_size()` in g1CollectedHeap.cpp I think we should lift it out to a helper that can be used by both G1 and Parallel. Not entirely sure where the best location for such helper is, but a static function `ReservedSpace::actual_page_size(ReservedSpace)` could work.